src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1030,19 +1030,22 @@ void uct_dc_mlx5_iface_cleanup_fc_ep(uct_dc_mlx5_iface_t *iface)
 
 ucs_status_t uct_dc_mlx5_iface_fc_grant(uct_pending_req_t *self)
 {
-    uct_rc_pending_req_t *freq = ucs_derived_of(self, uct_rc_pending_req_t);
-    uct_dc_mlx5_ep_t *ep       = ucs_derived_of(freq->ep, uct_dc_mlx5_ep_t);
-    uct_rc_iface_t *iface      = ucs_derived_of(ep->super.super.iface,
-                                                uct_rc_iface_t);
+    uct_dc_fc_request_t *fc_req = ucs_derived_of(self, uct_dc_fc_request_t);
+    uct_dc_mlx5_ep_t *ep        = ucs_derived_of(fc_req->super.ep,
+                                                 uct_dc_mlx5_ep_t);
+    uct_dc_mlx5_iface_t *iface  = ucs_derived_of(ep->super.super.iface,
+                                                 uct_dc_mlx5_iface_t);
```

Comment:
you can take rc iface here, like it was before.
then you can make simpler assert (like it was done before as well:)
`ucs_assert_always(iface->config.fc_enabled);`

Reply:
good catch, done